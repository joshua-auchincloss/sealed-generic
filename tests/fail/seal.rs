use sealed_generic::SealedGeneric;

#[derive(SealedGeneric)]
#[define(
    sealed,
    types(ty = i32),
)]
pub struct SomeGeneric<T: SealedSomeGeneric> {
    value: T,
}

fn main() {
    SomeGeneric::<String> {
        value: "test".into(),
    };
}
