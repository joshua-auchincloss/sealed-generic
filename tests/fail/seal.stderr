error[E0277]: the trait bound `String: SealedSomeGeneric` is not satisfied
  --> tests/fail/seal.rs:13:36
   |
13 |     SomeGeneric::<String> { value: "".into() };
   |                                    ^^^^^^^^^ the trait `SealedSomeGeneric` is not implemented for `String`
   |
   = help: the trait `SealedSomeGeneric` is implemented for `i32`
note: required by a bound in `SomeGeneric`
  --> tests/fail/seal.rs:8:27
   |
8  | pub struct SomeGeneric<T: SealedSomeGeneric> {
   |                           ^^^^^^^^^^^^^^^^^ required by this bound in `SomeGeneric`
